content/blog/2025-04-10-fastest-tpch-generator.md

```diff
@@ -0,0 +1,614 @@
+---
+layout: post
+title: tpchgen-rs World’s fastest open source TPC-H data generator, written in Rust
+date: 2025-04-10
+author: Andrew Lamb, Achraf B, and Sean Smith
+categories: [performance]
+---
+
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+<style>
+/* Table borders */
+table, th, td {
+  border: 1px solid black;
+  border-collapse: collapse;
+}
+th, td {
+  padding: 3px;
+}
+</style>
+
+3 members of the [Apache DataFusion] community used Rust and open source
+development to build [tpchgen-rs], a fully open TPC-H data generator over 10x
+faster than any other implementation  we know of.
+
+It is now possible to create the TPC-H SF=100 dataset in 72.23 seconds (1.4 GB/s
+😎) on a Macbook Air M3 with 16GB of memory, compared to the classic `dbgen`
+which takes 30 minutes<sup>1</sup> (0.05GB/sec). On the same machine, it takes less than
```

Comment:
throughput 1.4GB/s vs 0.05GB/sec = 28x

Reply:
That is a good point -- the 10x is a conservative estimate.

It @clflushopt 's measurements (2m vs 44m is also 22x 🤔 )

The 10x is the smallest improvement on the gcp measurements (where the machine has more resources): https://docs.google.com/spreadsheets/d/14qTHR5zgqXq4BkhO1IUw2BPwBUIOqMXLZ2fUyOaPflI/edit?gid=0#gid=0

I'll update the text to say over 20x faster